spec/support/macros.rb

```diff
@@ -0,0 +1,20 @@
+def set_current_user
+  thupten = Fabricate(:user)
+  session[:user_id] = thupten.id
+end
+
+def current_user
+  User.find(session[:user_id])
+end
+
+def clear_current_user
+  session[:user_id] = nil
+end
+
+def sign_in(a_user=nil)
+  user = a_user || Fabricate(:user)
+  visit ('/sign_in')
```

Comment:
When you use braces there are no spaces allowed between method name and the left brace. This can cause issues!